File: strong_password_python/strong_password.py
import re
from typing import Final


def is_password_strong(password):
    password_size: Final = r".{8,}"
    one_lowercase: Final = r"[a-z]"
    one_uppercase: Final = r"[A-Z]"
    one_digit: Final = r"\d"
    special_characters: Final = r"[!\?#$%&'()*+,-./:;<=>?@[\]^_`{|}~]"

    if re.search(password_size, password) is None:
        print('Password must have at least eight characters')
        return False

    elif re.search(one_lowercase, password) is None:
        print('Password must have at least one lowercase character')
        return False

    elif re.search(one_uppercase, password) is None:
        print('Password must have at least one uppercase character')
        return False

    elif re.search(one_digit, password) is None:
        print('Password must have at least one digit')
        return False

    elif re.search(special_characters, password) is None:
        print('Password must have at least one special character')
        return False

    password_strength(password)

    return True


def password_strength(password):
    if 8 <= len(password) < 12:
        print('Week Password')

    elif 12 <= len(password) < 14:
        print('Medium Password')
    else:
        print('Strong Password')


if __name__ == '__main__':
    print(is_password_strong('Abdallah@123456'))

File: strong_password_dart/strong_password.dart
bool isPasswordStrong(String password) {
  final passwordSize = RegExp(r'.{8,}');
  final oneLowercase = RegExp(r'[a-z]');
  final oneUppercase = RegExp(r'[A-Z]');
  final oneDigit = RegExp(r'\d');
  final specialCharacter = RegExp(r'\W');

  if (!passwordSize.hasMatch(password)) {
    print('Password must have at least eight characters long.');

    return false;
  } else if (!oneLowercase.hasMatch(password)) {
    print('Password must have at least one lower case character.');

    return false;
  } else if (!oneUppercase.hasMatch(password)) {
    print('Password must have at least one upper case character.');

    return false;
  } else if (!oneDigit.hasMatch(password)) {
    print('Password must have at least one digit.');

    return false;
  } else if (!specialCharacter.hasMatch(password)) {
    print('Password must have at least one special character.');

    return false;
  }

  print(passwordStrength(password));

  return true;
}

String passwordStrength(String password) {
  if (password.length >= 8 && password.length < 12) {
    return 'Week Password';
  } else if (password.length >= 12 && password.length < 14) {
    return 'Medium Password';
  }

  return "Strong Password";
}

void main() {
  print(isPasswordStrong('Abdallah@123456'));
}
